astropy/io/votable/converters.py

```diff
@@ -346,23 +336,28 @@ def output(self, value, mask):
 
     def _binparse_var(self, read):
         length = self._parse_length(read)
-        return read(length), False
+        return read(length).decode('ascii'), False
```

Comment:
Note: This no longer choose between `'ascii'` and `'utf-8'` based on `'verify'` setting. This one is stricter, but I suspect it wouldn't be hard to add it back in, though the VO table standard did say `char` is ASCII. 🤷‍♀

Reply:
With the updated commit, I changed this behavior to allow non-ASCII characters for FIELDs with `datatype="char"` in TABLEDATA serializations, but with a new warning message.  Whether all ASCII or not, values for those FIELDs are stored in the astropy Table as strings.

Reply:
It seems to me that @pllim's comment is still valid; but I'm not familiar with this code so maybe I'm misreading it. As in, putting utf-8 text in a char field and setting 'verify' to 'ignore' seems to raise an exception, but I've not actually tried this. I'm in favor of allowing utf-8, even if it is not in the official standard.

Reply:
Hi @hugobuddel .  Thanks for the taking a look at this.

The context here a is a little misleading.  You are correct that in the `_binparse_var()` method shown above, non-ASCII characters will cause an exception.  This occurs for the `BINARY` and `BINARY2` serializations when any of the byte values are outside the range 0-127.  I think that behavior is probably OK since values outside that range may just as likely represent corrupt binary data as opposed to deliberate characters in the extended ASCII range (128-255), but I could go either way.  I should (and will) at least add a test case to demonstrate that as the expected behavior.

My comment above though was referring to the `parse()` method that is just above `_binparse_var()`.  `parse()` handles the TABLEDATA serialization (in which non-ASCII unicode might most easily sneak in).  In that case a `try` clause catches the decode exception and calls `vo_warn()` with a new warning message.  `vo_warn()` looks at the `verify` value to decide whether to show the message.

Reply:
Thanks for the clarification @tomdonaldson . Adding tests would indeed help, because that's where I looked to see whether my interpretation of the code was correct, but it is clear now.

I'm personally not using the BINARY/BINARY2 serializations (only FITS or TABLEDATA), so I'll let others decide. (At least, I'm not writing them; no idea whether I read them from other sources.)